tests/RobotFramework/tests/plugin_apt/install_overwrite_config_files.robot

```diff
@@ -0,0 +1,52 @@
+*** Settings ***
+Resource            ../../resources/common.resource
+Library             ThinEdgeIO
+Library    Cumulocity
+
+Suite Setup         Custom Setup
+Suite Teardown      Get Logs
+
+Test Tags           theme:software    theme:plugins
+
+
+*** Test Cases ***
+Install packages without overwriting config files
+    Execute Command    tedge config set apt.dpk.options.config keepold
+    # install package v1
+    Execute Command   /etc/tedge/sm-plugins/apt install /setup/sampledeb_1.0.0_all.deb  
+    # Modify the config file
+    Execute Command    echo "Updated the config" | sudo tee -a /etc/sampledeb.cfg
+
+    # install package v2
+    Execute Command   /etc/tedge/sm-plugins/apt install /setup/sampledeb_2.0.0_all.deb  
+    # Assert to make sure installation of newer version did not update the config file
+    ${output}=    Execute Command    cat /etc/sampledeb.cfg
+    Should Contain    ${output}    conf 1.0
+    Should Contain    ${output}    Updated the config
+    ThinEdgeIO.Remove Package Using APT    sampledeb
+    Execute Command   rm /etc/sampledeb.cfg
+
+
+Install packages overwrite config files
+    Execute Command    tedge config set apt.dpk.options.config keepnew
+    # install package v1
+    Execute Command   /etc/tedge/sm-plugins/apt install /setup/sampledeb_1.0.0_all.deb  
+    # Modify the config file
+    Execute Command    echo "Updated the config" | sudo tee -a /etc/sampledeb.cfg
+
+    # install package v2
+    Execute Command   /etc/tedge/sm-plugins/apt install /setup/sampledeb_2.0.0_all.deb  
+    # Assert to make sure installation of newer version did not update the config file
+    ${output}=    Execute Command    cat /etc/sampledeb.cfg
+    Should Contain    ${output}    conf 2.0
+    ThinEdgeIO.Remove Package Using APT    sampledeb
+    Execute Command   rm /etc/sampledeb.cfg    
+
+*** Keywords ***
+Custom Setup  
+    ${DEVICE_SN}=    Setup
+    Set Suite Variable    $DEVICE_SN
+    Device Should Exist    ${DEVICE_SN}
+    Service Health Status Should Be Up    tedge-mapper-c8y
```

Comment:
You don't need this as you're interfacing with the `apt` sm-plugin directly and not using the cloud.

Reply:
removed